File: 3. Longest Substring Without Repeating Characters.py
# Problem Link: https://leetcode.com/problems/longest-substring-without-repeating-characters/
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_set = set()

        i, j = 0, 0
        length = len(s)
        max_length = 0
        while j < length:
            if s[j] in char_set:
                char_set.remove(s[i])
                i += 1
            else:
                char_set.add(s[j])
                j += 1
                max_length = max(len(char_set), max_length)

        return max_length

File: C#/3. Longest Substring Without Repeating Characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        HashSet<int> charSet = new HashSet<int>();
        int i = 0,  j = 0, maxLength = 0;
        
        while(j < s.Length){
            if(charSet.Contains(s[j])){
                charSet.Remove(s[i]);
                i++;
            }
            else {
                charSet.Add(s[j]);
                j++;
                maxLength = Math.Max(charSet.Count, maxLength);
            }
        }
        
        return maxLength;
    }
}